src/history.cpp

```diff
@@ -170,4 +171,60 @@ namespace mamba
         // }
         return map;
     }
+
+    void History::add_entry(const std::vector<History::UserRequest> user_request)
+    {
+        std::ofstream out;
+        out.open(m_history_file_path, std::ios_base::app);
+        if (out.fail())
+        {
+            throw std::runtime_error("Couldn't open file: " + m_history_file_path.string());
+        }
+        else
+        {
+            for (auto request : user_request)
+            {
+                out << "==> " << request.date << " <==" << std::endl;
+                out << "# cmd: " << request.cmd << std::endl;
+                out << "# conda version: " << request.conda_version << std::endl;
+                for (auto unlink_dist : request.unlink_dists)
+                {
+                    out << unlink_dist << std::endl;
+                }
+                for (auto link_dist : request.link_dists)
+                {
+                    out << link_dist << std::endl;
+                }
+                if (request.update.size() > 0)
+                {
+                    out << "# update specs: [";
+                    std::string update_string;
+                    for (auto u : request.update)
+                    {
+                        update_string += "'" + u + "',";
+                    }
+                        update_string.back() = ']';
+                        out << update_string << std::endl;
+                }
+                if (request.remove.size() > 0)
+                {
+                    out << "# remove specs: [";
+                    for (auto r : request.remove)
```

Comment:
Maybe move this repeated code to write a "Python list" to a lambda function that converts a vector of strings to this pythonic std::string?

Reply:
Can you explain a bit more?
What would this lambda function do? It'd modify the `u`s from the `for` and format them the right way?

Reply:
I was thinking you can make a lambda funtion taht takes two inputs: 

1. "remove", "update", "neuter"
2. std::vector<string>

and then the lambda spits out a std::string like `# XXXXXX specs: [ '....', '...', ...]`

so something like

```
auto to_string_list = [](const std::string& action, const std::vector<std::string>& specs) -> std::string {
    make_my_string
}
```

what do you think? would that work?

Reply:
It makes sense.
Much more fancy
gonna implement that
thanks Wolf